### PR TITLE
feat: [CI-3333]: Scala/Kotlin yaml change

### PR DESCRIPTION
**Pull request by jamie-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37787)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37787)

File: commons/go/lib/utils/utils.go [hunk 1]
```diff
@@ -133,7 +133,20 @@ func ParseJavaNode(filename string) (*Node, error) {
 		parts := strings.Split(filename, "/")
 		node.File = parts[len(parts)-1]
 		node.Lang = LangType_JAVA
-	} else {
+	} else if strings.HasSuffix(filename, ".scala") {
+		// Cannot make any assumption from scala/kotlin file path, return generic source node for now
+		node.Lang = LangType_JAVA
```
Comment (hemanthmantri) [inline commons/go/lib/utils/utils.go:138]: Why aren't we keeping this separate (LangType_SCALA)? Same with Kotlin.

File: product/ci/addon/testintelligence/java/sbt.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Harness Inc. All rights reserved.
```
Comment (hemanthmantri) [inline product/ci/addon/testintelligence/java/sbt.go:1]: 2022

File: commons/go/lib/utils/utils.go (hunk 1 above)
Comment (jamie-harness) [reply commons/go/lib/utils/utils.go:138]: It will cause a lot of duplication... Anything JVM I think it should be around there

**Review by hemanthmantri — APPROVED** [review]
:lgtm:

> Reviewed 8 of 9 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37787#-:-NCa_eul7TXK-ISDGy0H:bnfp4nl)* status: Waiting on team(s): sto.

**Review by williamwissemann — APPROVED** [review]

**Pull request merged**